musicbot/bot.py

```diff
@@ -2423,14 +2423,15 @@ def check(message):
         """
 
         if message.attachments:
-            thing = message.attachments[0]['url']
+            thing = message.attachments[0].url
         elif url:
             thing = url.strip('<>')
         else:
             raise exceptions.CommandError("You must provide a URL or attach a file.", expire_in=20)
 
         try:
-            with aiohttp.Timeout(10):
+            timeout = aiohttp.ClientTimeout(total=10)
+            async with aiohttp.ClientSession(timeout=timeout):
                 async with self.aiosession.get(thing) as res:
```

Comment:
...when the next line uses a **different** ClientSession to make a call? You're wrapping one ClientSession's get request inside of another ClientSession? I saw on #1678 that you said you found http://docs.aiohttp.org/en/latest/client_quickstart.html#timeouts and added some code to fix it.

Then why aren't you just overriding it for this one `.get` call?
![](https://i.imgur.com/jscPwWr.png)